### PR TITLE
Align with Commonalities subscription-model by using sink and sinkCredentials

### LINKED ISSUE #38: Validate API code and align with commonalities (meta-release item 1-2-3)
- Confirm API definition
-  Design guidelines from Commonalities applied (TBC)
- Guidelines from ICM applied  (TBC)

### PR DESCRIPTION
**Pull request by jgarciahospital**
#### What type of PR is this?

Add one of the following kinds:

* correction
* enhancement/feature
* cleanup
* documentation

#### What this PR does / why we need it:

Fixing small code and documentation typos and including  alignment of "$.sink" and "$.sinkCredentials" properties.

#### Which issue(s) this PR fixes:

Fixes #38

### CODE REVIEW

**Comment by rartych** [comment]
In my view it is not the subscription, but asynchronous response - therefore I opened the issue: https://github.com/camaraproject/Commonalities/issues/297
If we use subscription guidelines the the response should be in Cloudevents format.
Since there is no guidelines I am also fine with the proposed changes.

**Review by rartych — APPROVED** [review]

File: code/API_definitions/population-density-data.yaml [hunk 1]
```diff
@@ -595,11 +689,11 @@ components:
...
         application/json:
           schema:
-            $ref: "#/components/schemas/ErrorInfo"
+            $ref: '#/components/schemas/ErrorInfo'
           example:
             status: 400
-            code: "INVALID_ARGUMENT"
-            message: "Client specified an invalid argument, request body or query param"
+            code: 'INVALID_ARGUMENT'
+            message: 'Client specified an invalid argument, request body or query param'
```
Comment (bigludo7) [inline code/API_definitions/population-density-data.yaml:696]: What about adding also to be fully aligned with commonalities for the subscription?

```
           GENERIC_400_INVALID_PROTOCOL:
              value:
                status: 400
                code: "INVALID_PROTOCOL"
                message: "Only HTTP is supported"
            GENERIC_400_INVALID_CREDENTIAL:
              value:
                status: 400
                code: "INVALID_CREDENTIAL"
                message: "Only Access token is supported"
            GENERIC_400_INVALID_TOKEN:
              value:
                status: 400
                code: "INVALID_TOKEN"
                message: "Only bearer token is supported"`

**Review by bigludo7 — COMMENTED** [review]
Proposal to add 3 400 error and that will be good for me.
I saw @rartych comment - agreed that we can align the event itself after the meta release as ti is probably too late to adjust now.

File: code/API_definitions/population-density-data.yaml (hunk 1 above)
Comment (jgarciahospital) [reply code/API_definitions/population-density-data.yaml:710]: hope it's solved now, added (except Invalid_protocol, not sure where it comes from) and also added generic missing 410

**Review by bigludo7 — APPROVED** [review]
LGTM

**Review by sachinvodafone — APPROVED** [review]

**Pull request merged**